### PR TITLE
Adds no-useless-path-segments documentation.

### LINKED ISSUE #1032: Document no-useless-path-segments
cc @graingert 
re #912

### PR DESCRIPTION
**Pull request by manovotny**
Fixes #1032

### CODE REVIEW

**Comment by ljharb** [comment]
cc @graingert

File: docs/rules/no-useless-path-segments.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+/**
+ *  in my-project/app.js
+ */
+
+import "./../pages/about.js"; // should be "./pages/about.js"
+import "./../pages/about"; // should be "./pages/about"
+import "../pages/about.js"; // should be "./pages/about.js"
+import "../pages/about"; // should be "./pages/about"
+import "./pages//about"; // should be "./pages/about"
+import "./pages/"; // should be "./pages"
```
Comment (graingert) [inline docs/rules/no-useless-path-segments.md]: might be worth adding a note about:

`import "../"` -> `import "./"`

File: docs/rules/no-useless-path-segments.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+The following patterns are considered problems:
+
+```js
+/**
+ *  in my-project/app.js
+ */
+
+import "./../pages/about.js"; // should be "./pages/about.js"
```
Comment (graingert) [inline docs/rules/no-useless-path-segments.md]: shouldn't this be `"./../my-project/pages/about.js" // should be "./pages/about.js"`

**Comment by ljharb** [comment]
Indeed; a follow up would be appreciated.

**Pull request merged**